spec/models/preserved_copy_spec.rb

```diff
@@ -284,34 +306,34 @@
       end
 
       describe '.by_moab_storage_root_name' do
-        let(:pcs_ordered_by_query1) { described_class.fixity_check_expired.by_moab_storage_root_name(ms_root.name) }
-        let(:pcs_ordered_by_query2) { described_class.fixity_check_expired.by_moab_storage_root_name(ms_root2.name) }
+        let(:random_pcs_by_query1) { described_class.fixity_check_expired.by_moab_storage_root_name(ms_root.name) }
+        let(:random_pcs_by_query2) { described_class.fixity_check_expired.by_moab_storage_root_name(ms_root2.name) }
 
-        it 'returns PreservedCopies with nils first, then old to new timestamps, only for the chosen storage root' do
-          expect(pcs_ordered_by_query1).to eq [pc, checked_before_threshold_pc1]
-          expect(pcs_ordered_by_query2).to eq [checked_before_threshold_pc2]
+        it 'returns PreservedCopies < given date only for the chosen storage root (not ordered by last_version_audit)' do
+          expect(random_pcs_by_query1.sort).to eq [pc, checked_before_threshold_pc1]
+          expect(random_pcs_by_query2.sort).to eq [checked_before_threshold_pc2]
         end
         it 'returns no PreservedCopies with timestamps indicating fixity check in the last week' do
-          expect(pcs_ordered_by_query1).not_to include recently_checked_pc1
-          expect(pcs_ordered_by_query2).not_to include recently_checked_pc2
+          expect(random_pcs_by_query1).not_to include recently_checked_pc1
+          expect(random_pcs_by_query2).not_to include recently_checked_pc2
         end
       end
 
       describe '.by_storage_location' do
-        let!(:pcs_ordered_by_query1) do
+        let!(:random_pcs_by_query1) do
           described_class.fixity_check_expired.by_storage_location('spec/fixtures/storage_root01/sdr2objects')
         end
-        let!(:pcs_ordered_by_query2) do
+        let!(:random_pcs_by_query2) do
           described_class.fixity_check_expired.by_storage_location('spec/fixtures/storage_root02/sdr2objects')
         end
 
-        it 'returns PreservedCopies with nils first, then old to new timestamps, only for the chosen storage root' do
-          expect(pcs_ordered_by_query1).to eq [pc, checked_before_threshold_pc1]
-          expect(pcs_ordered_by_query2).to eq [checked_before_threshold_pc2]
+        it 'returns PreservedCopies < given date only for the chosen storage root (not ordered by last_version_audit)' do
+          expect(random_pcs_by_query1.sort).to eq [pc, checked_before_threshold_pc1]
+          expect(random_pcs_by_query2.sort).to eq [checked_before_threshold_pc2]
         end
         it 'returns no PreservedCopies with timestamps indicating fixity check in the last week' do
-          expect(pcs_ordered_by_query1).not_to include recently_checked_pc1
-          expect(pcs_ordered_by_query2).not_to include recently_checked_pc2
+          expect(random_pcs_by_query1).not_to include recently_checked_pc1
```

Comment:
Any help with variable names would be great.

Reply:
i think you could just drop the `random_` prefix and that'd be fine.  the var name as it is reads to me more like the set of things returned is random, as opposed to the ordering being non-deterministic (which i think will be even less obvious outside the context of this diff).